koocook_core/support/quantity.py

```diff
@@ -12,36 +13,39 @@ class Quantity:
     __slots__ = ('amount', 'unit')
 
     def __init__(self,
-                 amount: float,
+                 amount: Union[Fraction, int, float],
                  unit: Union[unit_.Unit, str]):
-        if isinstance(amount, float):
+        if isinstance(amount, Fraction):
             self.amount = amount
         else:
-            self.amount = float(amount)
+            self.amount = Fraction(amount)
         self.unit = unit_.get_unit(unit)
 
     def __str__(self):
         if self.amount == 1:
-            return '{:.0f} {}'.format(self.amount, self.unit.singular)
-        if self.amount.is_integer():
-            return '{:.0f} {}'.format(self.amount, self.unit.plural)
+            return '{} {}'.format(self.amount, self.unit.singular)
         return '{} {}'.format(self.amount, self.unit.plural)
 
     def get_db_str(self):
+        if self.unit.symbol:
+            return '{} {}'.format(self.amount, self.unit.symbol)
         return self.__str__()
 
-    @property
-    def not_a_unit(self):
-        return self.unit is unit_.SpecialUnit.NONE
-
     # Monkey patched
     def __len__(self):
         return len(str(self))
 
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        if other.amount == self.amount and other.unit == self.unit:
+            return True
+        return False
```

Comment:
You could instead replace those three lines with a single `return` statement.

Reply:
Accepted in 94deed6